wikimedia/wikipedia-ios
# [ALT TEXT] Display Article (Part 2)

**Phabricator:** https://phabricator.wikimedia.org/T370231

### Notes
This PR completes Article updates for displaying within the Experiment B flow. Here's the work that it performs:

- Brings over some more data from the image recommendation into the AltTextExperimentViewModel. This data is either needed for scrolling to the image or eventually for publishing.
- Sets up a different navigation bar for Article in this context.
- Hides edit pencils, disables links, and hides footer.
- Doesn't allow landscape
- Scrolls to the image after an article loads.
- Updates the article web view content inset so the entire article can be viewed when half sheet modal is displayed.

Note that this does not yet display after the article editor (Experiment C). That piece is blocked by https://phabricator.wikimedia.org/T370305.

### Test Steps
1. Fresh install app. Choose a target wiki in onboarding as a primary app language, or choose EN.
2. Go to developer settings and select ON "Enable Alt Text experiment". If you are testing against EN also select ON "Enable Alt Text experiment for English Wikipedia"
3. Go through image recommendations flow and publish an edit. Note this will be posting to EN Wiki, so confirm it's a good image.
4. (If you are in the lucky 50%) You will see the alt text prompt modal. Choose to add alt text.
5. Article is then pushed on. Confirm you are scrolled to the image in the article. Confirm you cannot rotate. Confirm edit pencils and footer is hidden and links do nothing. Confirm the temporary half sheet modal doesn't hide the bottom of the article.

### Screenshots/Videos
![out](https://github.com/user-attachments/assets/ded2cb76-04cd-48a6-a0e8-34274a3fe4f3)
